contrib/mesos/pkg/scheduler/components/framework/framework.go

```diff
@@ -123,6 +125,20 @@ func New(config Config) Framework {
 					return false
 				}
 
+				// at most one executor id expected. More than one means that
+				// the given node is seriously in trouble.
+				if len(o.GetExecutorIds()) > 1 {
+					return false
+				}
+
+				// the executor id must match, otherwise the running executor
+				// is incompatible with the current scheduler configuration.
+				if len(o.GetExecutorIds()) == 1 {
+					if eid := o.GetExecutorIds()[0]; eid.GetValue() != config.ExecutorId.GetValue() {
+						return false
+					}
+				}
```

Comment:
instead of these `if`s and the nested `if` I think a switch would be more readable:
```
eids := len(o.GetExecutorIDs())

switch {
  eids > 1:
    return false

  eids == 1:
    if eid := ... {
      return false
    }
}
```